### PR TITLE
feat: clickhouse connector

### PR DESCRIPTION
**Pull request by raphaelvignes**
## Change Summary

New ClickHouse Connector

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable

### CODE REVIEW

File: toucan_connectors/clickhouse/clickhouse_connector.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                        f"""SELECT name FROM system.tables WHERE database = '{current_config["database"]}'"""
+                    )
+                    res = cursor.fetchall()
+                    available_tables = [table[0] for table in res]
+                    constraints['table'] = strlist_to_enum('table', available_tables)
+
+        return create_model('FormSchema', **constraints, __base__=cls).schema()
+
+
+class ClickhouseConnector(ToucanConnector):
```
Comment (piotch) [inline toucan_connectors/clickhouse/clickhouse_connector.py:84]: @raphaelvignes I think we should add a configuration option to support SSL wrapped TCP connection ("secure connection"). The protocol part of the url is not `clickhouse` in this case but `clickhouses` (with a S)

https://clickhouse-driver.readthedocs.io/en/latest/api.html#clickhouse_driver.Client.from_url

Comment (raphaelvignes) [reply toucan_connectors/clickhouse/clickhouse_connector.py:81]: Thanks, I added this option.

File: toucan_connectors/clickhouse/clickhouse_connector.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    )
+    port: int = Field(None, description='The listening port of your database server')
+    user: str = Field(..., description='Your login username')
+    password: SecretStr = Field('', description='Your login password')
+    ssl_connection: bool = Field(False, description='Create a SSL wrapped TCP connection')
+
+    def get_connection_url(self, *, database='default'):
+        if self.ssl_connection:
+            return f'clickhouses://{self.user}:{self.password.get_secret_value() if self.password else ""}@{self.host}:{self.port}/{database}'
+        return f'clickhouse://{self.user}:{self.password.get_secret_value() if self.password else ""}@{self.host}:{self.port}/{database}'
```
Comment (piotch) [inline toucan_connectors/clickhouse/clickhouse_connector.py:100]:
```suggestion
        proto = 'clickhouses' if self.ssl_connection else 'clickhouse'
        return f'{proto}://{self.user}:{self.password.get_secret_value() if self.password else ""}@{self.host}:{self.port}/{database}'
```

**Review by piotch — APPROVED** [review]

File: toucan_connectors/clickhouse/clickhouse_connector.py (hunk 2 above)
Comment (raphaelvignes) [reply toucan_connectors/clickhouse/clickhouse_connector.py:100]: Thx

**Pull request merged**